tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/CardinalityConstRule.scala

```diff
@@ -76,7 +76,7 @@ class CardinalityConstRule(rewriter: SymbStateRewriter) extends RewritingRule {
     }
 
     // create the inequality predicate
-    val witnesses = 1.to(threshold).map(pick).toList
+    val witnesses = 1.to(threshold).map(_ => pick()).toList
```

Comment:
```suggestion
    val witnesses = List.fill(threshold)(pick())
```

Reply:
I'd rather not do this. To understand its correctness, you'd have to both know that the parameter is call-by-name and that it's evaluated for each element. Neither is obvious from the snippet.

Reply:
If you want to emphasize that pick is called separately each time you can still do
```suggestion
    val witnesses = List.fill(threshold)(_ => pick())
```
instead of `to` + `map`